### PR TITLE
NSFS | NC | Health script | distinguish between temporary and persistent errors

### PR DESCRIPTION
**Pull request by naveenpaul1**
### Explain the changes
1. Heath script updated with error_type and for temporary errors multiple retries are added
2. Health issue with missing `accounts` and `buckets` folder in the config root path
3. Rsyslog start-up error fixed by removing unwanted default value configuration in noobaa_rsyslog.config

### Issues: Fixed #xxx / Gap #xxx
1. https://github.com/noobaa/noobaa-core/issues/7714
2. https://github.com/noobaa/noobaa-core/issues/7730

### Testing Instructions:
1. run the health script and verify the health response structure similar to below
```{
  "service_name": "nsfs",
  "status": "NOTOK",
  "memory": "88.6M",
  "error": {
    "error_code": "RSYSLOG_SERVICE_FAILED",
    "error_message": "RSYSLOG service is not started properly, Please verify the service with status command."
  },
  "checks": {
    "services": [
      {
        "name": "nsfs",
        "service_status": "active",
        "pid": "1204",
        "error_type": "PERSISTENCE"
      },
      {
        "name": "rsyslog",
        "service_status": "inactive",
        "pid": "0",
        "error_type": "PERSISTENCE"
      }
    ],
    "endpoint": {
      "endpoint_state": {
        "response": {
          "response_code": 200,
          "response_message": "Endpoint running successfuly."
        },
        "total_fork_count": 1,
        "running_workers": [
          "1"
        ]
      },
      "error_type": "TEMPORARY"
    },
    "accounts_status": {
      "invalid_accounts": [
        {
          "name": "naveen",
          "storage_path": "/tmp/nsfs_root_invalid/",
          "code": "STORAGE_NOT_EXIST"
        }
      ],
      "valid_accounts": [
        {
          "name": "naveen",
          "storage_path": "/tmp/nsfs_root"
        }
      ],
      "error_type": "PERSISTENCE"
    },
    "buckets_status": {
      "invalid_buckets": [
        {
          "name": "bucket1.json",
          "config_path": "/etc/noobaa.conf.d/buckets/bucket1.json",
          "code": "INVALID_CONFIG"
        },
        {
          "name": "bucket3",
          "storage_path": "/tmp/nsfs_root/bucket3",
          "code": "STORAGE_NOT_EXIST"
        }
      ],
      "valid_buckets": [
        {
          "name": "bucket2",
          "storage_path": "/tmp/nsfs_root/bucket2"
        }
      ],
      "error_type": "PERSISTENCE"
    }
  }
} ```

- [X] Doc added/updated
- [X] Tests added

### CODE REVIEW

File: src/cmd/health.js [hunk 1]
```diff
@@ -129,24 +134,61 @@ class NSFSHealth {
...
+      delete health.checks.buckets_status;
+    }
     return health;
   }
 
+  async get_endpoint_response() {
+    let endpoint_state;
+    try {
+      await P.retry({
+        attempts: config.NSFS_HEALTH_ENDPOINT_RETRY_COUNT,
```
Comment (romayalon) [inline src/cmd/health.js:174]: are these new properties? I don't see they were added to config.js or if they exists already

File: src/cmd/health.js [hunk 2]
```diff
@@ -257,24 +299,35 @@ class NSFSHealth {
...
-  const entries = await nb_native().fs.readdir(fs_context, this.get_config_path(config_root, type));
-  const config_files = entries.filter(entree => !native_fs_utils.isDirectory(entree) && entree.name.endsWith('.json'));
+  const config_root_type_path = this.get_config_path(config_root, type);
   const invalid_storages = [];
   const valid_storages = [];
+  //check for account and buckets dir paths
+  try {
+    await nb_native().fs.stat(fs_context, config_root_type_path);
+  } catch (err) {
+    dbg.log1(`Config root path not created yet`, config_root_type_path);
```
Comment (romayalon) [inline src/cmd/health.js:321]: config root might got deleted as well

File: src/cmd/health.js (hunk 1 above)
Comment (naveenpaul1) [reply src/cmd/health.js:174]: @romayalon missed this property after rebase :), Will add

**Review by romayalon — APPROVED** [review]

**Comment by romayalon** [comment]
@naveenpaul1 see the last comment, can we do it?

**Comment by naveenpaul1** [comment]
@romayalon @ps20renar  yes, can do a similar check depending on the user input.

**Pull request merged**